presto-main/src/main/java/io/prestosql/spi/expression/ConnectorExpressionTranslator.java

```diff
@@ -82,7 +82,7 @@ public Expression translate(ConnectorExpression expression)
             if (expression instanceof FieldDereference) {
                 FieldDereference dereference = (FieldDereference) expression;
 
-                RowType type = (RowType) expression.getType();
+                RowType type = (RowType) dereference.getTarget().getType();
```

Comment:
is it possible to test this?

also, cmt title is too long. Maybe just `Fix translation of FieldDereference to DereferenceExpression`.

Reply:
@findepi Fixed the message. Added a test in `TestConnectorExpressionTranslator`. I haven't added literals for the test since the translation for them would rely on how ExpressionAnalyzer and LiteralEncoder.